pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -162,8 +165,19 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, nil
 	}
 
+	// isStatusConditionPresent returns true when conditionType is present.
+	isStatusConditionPresent := func(conditions []metav1.Condition, conditionType string) bool {
```

Comment:
Instead of building your own function, you should use meta.IsStatusConditionPresentAndEqual